src/hypercorn/trio/tcp_server.py

```diff
@@ -48,28 +49,31 @@ async def run(self) -> None:
             socket = self.stream.socket
             ssl = False
 
+        def log_handler(e: Exception) -> None:
+            if self.config.log.error_logger is not None:
+                self.config.log.error_logger.exception("Internal hypercorn error")
+
         try:
-            client = parse_socket_addr(socket.family, socket.getpeername())
-            server = parse_socket_addr(socket.family, socket.getsockname())
-
-            async with TaskGroup() as task_group:
-                self._task_group = task_group
-                self.protocol = ProtocolWrapper(
-                    self.app,
-                    self.config,
-                    self.context,
-                    task_group,
-                    ssl,
-                    client,
-                    server,
-                    self.protocol_send,
-                    alpn_protocol,
-                )
-                await self.protocol.initiate()
-                await self._start_idle()
-                await self._read_data()
-        except OSError:
-            pass
```

Comment:
There's a minor behavioral change here where we used to not log `OSError`s, but now we do. Probably it's better to keep that, because `OSError`s are probably like "the other side hung up" or similar and not interesting? I think you can do it with like

`catch({OSError: ignore_handler, Exception: log_handler})`

Reply:
done